pkg/api/validation/validation.go

```diff
@@ -266,6 +266,28 @@ func validateLifecycle(lifecycle *api.Lifecycle) errs.ValidationErrorList {
 	return allErrs
 }
 
+// TODO(dchen1107): Move this along with other defaulting values
+func validatePullPolicyWithDefault(ctr *api.Container) errs.ValidationErrorList {
+	allErrors := errs.ValidationErrorList{}
+
+	if len(ctr.ImagePullPolicy) == 0 {
+		parts := strings.SplitN(ctr.Image, ":", 2)
+		// Check image tag
+		if len(parts) == 2 && parts[1] == "latest" {
```

Comment:
Re:
```
parts := strings.SplitN(ctr.Image, ":", 2)
```
Surely nothing good can happen if there is more than one colon in the image name? Perhaps it's worth raising an error for this case?